app/src/main/java/net/sourceforge/opencamera/sensorremote/RemoteRpcServer.java

```diff
@@ -0,0 +1,183 @@
+            long duration = Long.parseLong(imuRequestMatcher.group(2));
+            boolean wantAccel = Integer.parseInt(imuRequestMatcher.group(4)) == 1;
+            boolean wantGyro = Integer.parseInt(imuRequestMatcher.group(6)) == 1;
+
+            if (MyDebug.LOG) {
+                Log.d(TAG, "received IMU control request, duration = " + duration);
+            }
+            RemoteRpcResponse imuResponse = mRequestHandler.handleImuRequest(duration, wantAccel, wantGyro);
+
+            outputStream.println(imuResponse.toString());
+            if (MyDebug.LOG) {
+                Log.d(TAG, "IMU request file sent");
+            }
+            return;
+        }
+
+        // Video remote control API
+        if (msg.equals(mConfig.getProperty("VIDEO_START_REQUEST"))) {
+            outputStream.println(
+                    mRequestHandler.handleVideoStartRequest()
+            );
+        } else if (msg.equals(mConfig.getProperty("VIDEO_STOP_REQUEST"))) {
+            outputStream.println(
+                    mRequestHandler.handleVideoStopRequest()
+            );
+        } else if (msg.equals(mConfig.getProperty("GET_VIDEO_REQUEST"))) {
+            mRequestHandler.handleVideoGetRequest(outputStream);
+        }
+    }
+
+    @Override
+    public void run() {
+        // TODO: report hostname some other way
+        mContext.runOnUiThread(
+                () -> {
+                    try {
+                        new AlertDialog.Builder(mContext)
+                                .setTitle("Smartphone hostname")
+                                .setMessage("Hostname: " + getIPAddress())
+
+                                .setPositiveButton(android.R.string.yes, (dialog, which) -> {
+                                })
+                                .setIcon(android.R.drawable.ic_dialog_alert)
+                                .show();
+                    } catch (SocketException e) {
+                        e.printStackTrace();
+                    }
+
+                }
+        );
+
+        mIsExecuting.set(true);
+        if (MyDebug.LOG) {
+            Log.d(TAG, "waiting to accept connection from client...");
+        }
+        while (mIsExecuting.get()) {
+            try {
+                Socket clientSocket = mRpcSocket.accept();
+                clientSocket.setKeepAlive(true);
+                if (MyDebug.LOG) {
+                    Log.d(TAG, "accepted connection from client");
+                }
+                InputStream inputStream = clientSocket.getInputStream();
+                PrintStream outputStream = new PrintStream(clientSocket.getOutputStream());
+                BufferedOutputStream outputByte = new BufferedOutputStream(clientSocket.getOutputStream());
+                BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+
+                String inputLine;
+                while (!clientSocket.isClosed() && (inputLine = reader.readLine()) != null) {
```

Comment:
Advise: check the exit condition (`mIsExecuting`) here too. Server must shutdown immediately despite of the input length.